lib/node_modules/@stdlib/ndarray/base/every-by/test/test.2d.js

```diff
@@ -0,0 +1,1005 @@
+	var actual;
+	var xbuf;
+	var ord;
+	var sh;
+	var st;
+	var dt;
+	var o;
+	var x;
+
+	dt = 'complex128';
+	ord = 'column-major';
+	sh = [ 2, 2 ];
+	st = shape2strides( sh, ord );
+	o = strides2offset( sh, st );
+
+	xbuf = zeros( numel( sh )*2, 'float64' );
+	x = ndarray( dt, new Complex128Array( xbuf ), sh, st, o, ord );
+
+	actual = everyBy( [ x ], clbk );
+	t.strictEqual( actual, false, 'returns expected value' );
+
+	xbuf = ones( numel( sh )*2, 'float64' );
+	x = ndarray( dt, new Complex128Array( xbuf ), sh, st, o, ord );
+
+	actual = everyBy( [ x ], clbk );
+	t.strictEqual( actual, true, 'returns expected value' );
+
+	t.end();
+
+	function clbk( v ) {
+		return ( real( v ) !== 0.0 && imag( v ) !== 0.0 );
+	}
+});
+
+tape( 'the function tests whether every element in a 2-dimensional ndarray passes a test implemented by a predicate function (column-major, contiguous, negative strides, accessors)', function test( t ) {
+	var actual;
+	var xbuf;
+	var ord;
+	var sh;
+	var st;
+	var dt;
+	var o;
+	var x;
+
+	dt = 'complex128';
+	ord = 'column-major';
+	sh = [ 2, 2 ];
+	st = [ -1, -2 ];
+	o = strides2offset( sh, st );
+
+	xbuf = zeros( numel( sh )*2, 'float64' );
+	x = ndarray( dt, new Complex128Array( xbuf ), sh, st, o, ord );
+
+	actual = everyBy( [ x ], clbk );
+	t.strictEqual( actual, false, 'returns expected value' );
+
+	xbuf = ones( numel( sh )*2, 'float64' );
+	x = ndarray( dt, new Complex128Array( xbuf ), sh, st, o, ord );
+
+	actual = everyBy( [ x ], clbk );
+	t.strictEqual( actual, true, 'returns expected value' );
+
+	t.end();
+
+	function clbk( v ) {
+		return ( real( v ) !== 0.0 && imag( v ) !== 0.0 );
+	}
+});
+
+tape( 'the function tests whether every element in a 2-dimensional ndarray passes a test implemented by a predicate function (column-major, non-contiguous, same sign strides, accessors)', function test( t ) {
+	var actual;
+	var xbuf;
+	var ord;
+	var sh;
+	var st;
+	var dt;
+	var o;
+	var x;
+
+	dt = 'complex128';
+	ord = 'column-major';
+	sh = [ 2, 2 ];
+	st = [ 1, 4 ];
+	o = strides2offset( sh, st );
+
+	xbuf = zeros( 8*2, 'float64' );
+	x = ndarray( dt, new Complex128Array( xbuf ), sh, st, o, ord );
+
+	actual = everyBy( [ x ], clbk );
+	t.strictEqual( actual, false, 'returns expected value' );
+
+	xbuf = ones( 8*2, 'float64' );
+	x = ndarray( dt, new Complex128Array( xbuf ), sh, st, o, ord );
+
+	actual = everyBy( [ x ], clbk );
+	t.strictEqual( actual, true, 'returns expected value' );
+
+	t.end();
+
+	function clbk( v ) {
+		return ( real( v ) !== 0.0 && imag( v ) !== 0.0 );
+	}
+});
+
+tape( 'the function tests whether every element in a 2-dimensional ndarray passes a test implemented by a predicate function (column-major, non-contiguous, mixed sign strides, accessors)', function test( t ) {
+	var actual;
+	var xbuf;
+	var ord;
+	var sh;
+	var st;
+	var dt;
+	var o;
+	var x;
+
+	dt = 'complex128';
+	ord = 'column-major';
+	sh = [ 2, 2 ];
+	st = [ 4, -1 ];
```

Comment:
The strides are incorrect here.

Reply:
You should check your other similar PRs, as well, as I imagine this has been copy-pasted elsewhere.

Reply:
If this came from an existing set of tests, we should fix in those tests, as well.